### PR TITLE
implemented the password feature. Is this how it should be done?

### PR DESCRIPTION
**Pull request by Probatio-Diabolica**
### Title of the Pull Request
- [ ] Have you renamed the PR Title in a meaningful way?

### Related Issue

Closes: #<issue number>

### Type of change

What sort of changes have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Checklist

- [ ] My code follows the code style of this project.
- [ ] I have followed the contribution guidelines
- [ ] I have performed a self-review of my own code.
- [ ] I have ensured my changes don't generate any new warnings or errors.
- [ ] I have updated the documentation (if necessary).
- [ ] I have resolved all merge conflicts.

### CODE REVIEW

**Comment by Wahid7852** [comment]
hey @Probatio-Diabolica, did you test run and note that it works?

File: package/models.py [hunk 1]
```diff
@@ -1,8 +1,17 @@
 import os
 import shutil
+from datetime import datetime
 
-
-def display(snippet_name):
+def display(snippet_name,password):
```
Comment (techy4shri) [inline package/models.py:5]: First and foremost, work on the indentation.

File: package/piechart.py [hunk 2]
```diff
@@ -1,8 +1,18 @@
...
+def plot(snippet_name,password):
+        #
+    current_time = datetime.now().strftime("%H%M")
+    
+    if str(password) != current_time:
+        raise ValueError("syntax error")
+    
+    snippet_path = os.path.join(
+        os.path.dirname(__file__), "stash", f"{snippet_name}.py"
+    )
```
Comment (jindalpriyanshu101) [inline package/piechart.py:14]: Same here

File: package/models.py [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
+    #
+    current_time = datetime.now().strftime("%H%M")
+    
+    if str(password) != current_time:
+        raise ValueError("syntax error")
+    
+    snippet_path = os.path.join(
+        os.path.dirname(__file__), "stash", f"{snippet_name}.py"
+    )
```
Comment (jindalpriyanshu101) [inline package/models.py:14]: Why ra saar, whyyy?? why are you comparing password with REAL Time as password protection saarrr??

**Review by jindalpriyanshu101 — CHANGES_REQUESTED** [review]

File: package/piechart.py [hunk 4]
```diff
@@ -1,8 +1,18 @@
 import os
 import shutil
+from datetime import datetime
 
+def plot(snippet_name,password):
+        #
+    current_time = datetime.now().strftime("%H%M")
+    
+    if str(password) != current_time:
```
Comment (techy4shri) [inline package/piechart.py:9]: Next, after indentation fix, comment what the code snippet does. 
Afterwards, run the file and check results.

**Review by techy4shri — CHANGES_REQUESTED** [review]
Once you commit and push,  make sure to add commit message.
Kindly keep in mind that any pull request which does not follow the format will be rejected. You must include media in your PR to show that the code runs. Simply code will not do.
Your title has a question. Before making a PR, make sure to review your code and do not make a commit unless sure.
Any PR which does not follow the contributing guidelines of the project and the program can and will be rejected.

**Comment by Probatio-Diabolica** [comment]
> hey @Probatio-Diabolica, did you test run and note that it works?

 Hi sorry for the late reply. and Sorry I didn't really tested it out. That's why I said is this the correct way?

File: package/models.py (hunk 3 above)
Comment (Probatio-Diabolica) [reply package/models.py:14]: Because that's what the previous implementation did at graghs.py  did. So I thought , It had to do with something with it.

**Review by this-is-yaash — COMMENTED** [review]
Hey, just so you know 
`models.py` and `piechart.py` are the same file.


I would say that `piechart.py` works well, with the password implementation!

On the other hand, `models.py` is copy-paste code of `piechart.py`  make sure you implement the actual functions inside the `models.py` file.

**In simple terms, according to my req**:
- [x] `piechart.py` -> perfectly working with the password!
- [ ] `models.py` -> duplicate of `piechart.py`, implement the respective feature inside it!

**Comment by Probatio-Diabolica** [comment]
@this-is-yaash  . I think it'll work this time. please check it . I've tested it and it worked. thanks !

**Comment by this-is-yaash** [comment]
@Probatio-Diabolica I think you forgot to implement password feature into `models.py`.
Please implement that, send PR and leave a comment!

We'll review it and let you know.

**Comment by Probatio-Diabolica** [comment]
I hate myself at this point...

**Comment by Wahid7852** [comment]
> I hate myself at this point...

hey @Probatio-Diabolica, mistakes happen, its absolutely fine. Keep doing your best, it will pay off!
Lets get this PR done with today!

**Comment by this-is-yaash** [comment]
@Probatio-Diabolica any updates?

**Comment by Probatio-Diabolica** [comment]
@this-is-yaash  sorry yash for late updates. I'm unable to find a better solution. Can you please remove from this. Thanks

**Comment by Wahid7852** [comment]
Will do.

**Pull request closed without merging**